### PR TITLE
Jetpack Beta: Bug fixes and a new start

### PR DESCRIPTION
**Pull request by enejb**
I have updated the Jetpack beta to be more lean. 

There are still a few things to implement. 
- [x] Add Pressable beta install process.
- [x] More testing. 

cc: @lezama, @samhotchkiss 

Currently this PR does
Switching between master-stable and Beta versions. 
Updating to the right version as expected. 
Autoupdating is done the same way as Jetpack.

### CODE REVIEW

**Comment by samhotchkiss** [comment]
Per our convo today, I'm going to wait to see what we come up with next week!

**Comment by beaulebens** [comment]
I tried this out (paths have been `...`d to simplify):

```Warning: fopen(.../wp-content/plugins/jetpack-beta/jetpack-beta.php): failed to open stream: No such file or directory in .../wp-includes/functions.php on line 4694 Warning: fread() expects parameter 1 to be resource, boolean given in .../wp-includes/functions.php on line 4697 Warning: fclose() expects parameter 1 to be resource, boolean given in .../wp-includes/functions.php on line 4700```

I think it's misleading when you go to the plugin and the first thing you see is this:

screen shot 2017-01-23 at 12 30 29 pm

We should probably mention that it's "Tagged Beta and Stable Releases" or something, and show some indication of what I'm currently running. "Save my choice" isn't used anywhere else as a button label either, so we should probably just make that "Save Changes" to match most other buttons.

I switched to Bleeding Edge and was offered the ability to upgrade;

screen shot 2017-01-23 at 12 33 07 pm

Buttons never have `.` at the end, so we should remove that. Also, when I clicked the button, the text next to it (the message) jumped up for some reason (hard to capture in a screenshot). It gets moved vertically a few pixels (in Chrome). The message there is also a bit weird. We could just have it say "You have changed active Jetpack version. [Download & Install]" .

It might also be worth having a heading on the release notes box ("Latest Release Notes"?).

That being said, it looks like it works enough to let me switch between Tagged and Bleeding Edge. I haven't given it long enough to confirm the auto-update works yet.

**Comment by beaulebens** [comment]
I also noticed that I used this beta plugin to update to bleeding edge, then disabled the beta plugin, and now I'm still on that bleeding edge version. We might want to either automatically roll people back to a stable version, or at least offer an option within the Beta plugin to go back to the latest official release.

**Comment by samhotchkiss** [comment]
Looking really cool :) Can't wait for this to be ready to ship out!

File: admin/admin.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+(function () {
```
Comment (ebinnion) [inline admin/admin.js]: Throughout this file, you're missing some spaces between parentheses.

File: admin/admin.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        element.querySelector('.activate-branch').setAttribute('data-index', index);
+        section_index[index] = {
+            header: element.querySelector('.branch-card-header').textContent,
+            pr: element.getAttribute('data-pr'),
+            element: element
+        }
+    });
+
+    // Search input
+    search_input.addEventListener("keyup", function (event) {
```
Comment (ebinnion) [inline admin/admin.js]: Since quotes instead of double on `keyup`.

File: admin/admin.js (hunk 1 above)
Comment (ebinnion) [inline admin/admin.js]: Since we're using vanilla JS, we should be sure to test this cross browser and make sure we're good back to IE10.

**Comment by ebinnion** [comment]
Some functionality feedback:

- When I tried to find the `update/gulp-minify-js`, it didn't exist. That's probably because it's an old branch though
- Choosing the RC mentions Jetpack 4.8, but not the point releases we did to fix 4.8. Just something to consider
- When on bleeding edge, it'd be cool if it gave us an idea of the latest commits to go in
- I dig the "Jetpack Beta" navigation thing in the top bar, but the "Running Release Candiate" felt like I should be able to click it for some reason. Maybe we should style that differently.
    screen shot 2017-04-19 at 10 55 32 am
- Text spacing feels inconsistent
    screen shot 2017-04-19 at 11 00 32 am
- I turned on auto updates and pushed a new master-stable version. How long does it take to auto update?
- Can we add a description of what bleeding edge, latest stable, and release candidate mean?

**Comment by ebinnion** [comment]
Jetpack beta menu item doesn't show when WordPress.com toolbar is active.

screen shot 2017-04-19 at 11 07 02 am

**Comment by ebinnion** [comment]
The beta bug report link in the beta plugin points to Github, but in the Jetpack dashboard, we prompt users to give us feedback at https://jetpack.com/contact-support/beta-group/.

screen shot 2017-04-19 at 11 08 50 am
screen shot 2017-04-19 at 11 08 16 am

**Comment by enejb** [comment]
Thanks for the feedback Eric. 

> I turned on auto updates and pushed a new master-stable version. How long does it take to auto update?

It takes a while. :( Autoupdates happen every 12 hours. The new build (of master stable) only after Travis passes and there is a transient that stores the manifest data for 15 minutes.

**Comment by jeherve** [comment]
> The beta bug report link in the beta plugin points to Github, but in the Jetpack dashboard, we prompt users to give us feedback at https://jetpack.com/contact-support/beta-group/.

Fixed in 881ec11a91957edccc9e387c7d0e6e248359da9d

File: admin/main.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	<div class="dops-foldable-card is-expanded has-expanded-summary dops-card is-compact">
+		<div class="dops-foldable-card__header has-border">
+				<span class="dops-foldable-card__main">
+					<div class="dops-foldable-card__header-text">
+						<div class="dops-foldable-card__header-text"><?php _e( 'To Test', 'jetpack-beta' ); ?></div>
+					</div>
+				</span>
+		</div>
+		<div class="dops-foldable-card__content">
+			<?php echo $to_test ; ?>
```
Comment (lezama) [inline admin/main.php]: should we `esc_html`?

Comment (jeherve) [inline admin/main.php]: Since we can render markdown there, it may include some HTML. I don't think we can use `esc_html` here.

**Pull request merged**